src/test/kotlin/org/rust/ide/inspections/lints/RsRedundantSemicolonsInspectionTest.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsRedundantSemicolonsInspectionTest : RsInspectionsTestBase(RsRedundantSemicolonsInspection::class) {
+    fun `test single redundant semicolon`() = checkByText("""
+        fn main() {
+            struct A;
+            struct B;<weak_warning descr="Unnecessary trailing semicolon">;</weak_warning>
+        }
+    """)
+
+    fun `test multiple redundant semicolons on one line`() = checkByText("""
+        fn main() {
+            struct A;<weak_warning descr="Unnecessary trailing semicolons">; ;</weak_warning>
+            {<weak_warning descr="Unnecessary trailing semicolons">;;;</weak_warning>}
+        }
+    """)
+
+    fun `test multiple redundant semicolon on many lines`() = checkByText("""
+        fn main() {
+            struct A;<weak_warning descr="Unnecessary trailing semicolons">;
+
+            ;
+
+            ;</weak_warning>
+
+            {<weak_warning descr="Unnecessary trailing semicolons">;
+            ;
+            ;</weak_warning>}
+        }
+    """)
+
+    fun `test single fix`() = checkFixByText("Remove unnecessary trailing semicolons",
+    """
```

Comment:
Looks like this file was formatted automatically by IDEA because usually we put `"""` on the first line and then `""", """` together on the same line